specs/CDPSessionIdSupport.md

```diff
@@ -0,0 +1,328 @@
+                LogConsoleLogMessage(eventSource, eventDetails);
+                return S_OK;
+            })
+            .Get(),
+        &m_consoleAPICalledToken));
+    receiver.reset();
+  
+    // Track Target and session info via CDP events.
+    CHECK_FAILURE(
+        m_webView->GetDevToolsProtocolEventReceiver(L"Target.attachedToTarget", &receiver));
+    CHECK_FAILURE(receiver->add_DevToolsProtocolEventReceived(
+        Callback<ICoreWebView2DevToolsProtocolEventReceivedEventHandler>(
+            [this](
+                ICoreWebView2* sender,
+                ICoreWebView2DevToolsProtocolEventReceivedEventArgs* args) -> HRESULT
+            {
+                // A new target is attached, add its info to maps.
+                wil::unique_cotaskmem_string jsonMessage;
+                CHECK_FAILURE(args->get_ParameterObjectAsJson(&jsonMessage));
+                std::wstring sessionId = GetJSONStringField(jsonMessage.get(), L"sessionId");
+                std::wstring targetId = GetJSONStringField(jsonMessage.get(), L"targetId");
+                m_devToolsSessionMap[sessionId] = targetId;
+                std::wstring type = GetJSONStringField(jsonMessage.get(), L"type");
+                std::wstring url = GetJSONStringField(jsonMessage.get(), L"url");
+                m_devToolsTargetDescriptionMap[targetId] = type + L"," + url;
+                wil::com_ptr<ICoreWebView2_10> webview2 = m_webView.try_query<ICoreWebView2_10>();
+                if (webview2)
+                {
+                    // Auto attach to targets further created from this target.
+                    webview2->CallDevToolsProtocolMethodForSession(
+                        sessionId.c_str(), L"Target.setAutoAttach",
+                        LR"({"autoAttach":true,"waitForDebuggerOnStart":false,"flatten":true})",
+                        nullptr);
+                    // Enable Runtime events for the target to receive Runtime.consoleAPICalled from it.
+                    webview2->CallDevToolsProtocolMethodForSession(
+                        sessionId.c_str(), L"Runtime.enable", L"{}", nullptr);
+                }
+                return S_OK;
+            })
+            .Get(),
+        &m_targetAttachedToken));
+    receiver.reset();
+    CHECK_FAILURE(
+        m_webView->GetDevToolsProtocolEventReceiver(L"Target.detachedFromTarget", &receiver));
+    CHECK_FAILURE(receiver->add_DevToolsProtocolEventReceived(
+        Callback<ICoreWebView2DevToolsProtocolEventReceivedEventHandler>(
+            [this](
+                ICoreWebView2* sender,
+                ICoreWebView2DevToolsProtocolEventReceivedEventArgs* args) -> HRESULT
+            {
+                // A target is detached, remove it from the maps.
+                wil::unique_cotaskmem_string jsonMessage;
+                CHECK_FAILURE(args->get_ParameterObjectAsJson(&jsonMessage));
+                std::wstring sessionId = GetJSONStringField(jsonMessage.get(), L"sessionId");
```

Comment:
Will this field be present even on downlevel WebViews?

Reply:
This field is part of the CDP method result and presents on downlevel WebViews, just that we could not make use of it previously.

Reply:
Same as previous comment.